tests/integration/functional/test_client.py

```diff
@@ -964,3 +965,134 @@ def test_deleting_deployments(clean_client):
 
     with pytest.raises(KeyError):
         clean_client.get_deployment(str(response.id))
+
+
+class ClientCrudTestConfig(BaseModel):
+    entity_name: str
+    create_args: Dict[str, Any] = {}
+    get_args: Dict[str, Any] = {}
+    update_args: Dict[str, Any] = {}
+    delete_args: Dict[str, Any] = {}
+
+
+crud_test_configs = [
+    ClientCrudTestConfig(
+        entity_name="user",
+        create_args={"name": "user_name"},
+        update_args={"updated_name": "updated_user_name"},
```

Comment:
I love this generalized way of testing the client !

Could you please use some way to randomize the names of the created resources, similar to what we do in the zen_store generic CRUD tests ? As these tests will also be run on shared CI environments, they will trample over one-another. Using random names also makes it easier to run these tests in parallel in the future, or in another dimension.